[#775] Upgrade to node 24

daenamkim opened the pull request:
Resolves #775

daenamkim committed:
update nvmrc

daenamkim committed:
update package-lock.json

tawago commented:
Probably not specific to this PR but I keep getting below and am stuck 😅

```bash
$ run deploy:greeter

> sample@1.0.0 deploy:greeter
> HARDHAT_NETWORK=development npx hardhat run scripts/deploy-greeter.ts

WARNING: You are currently using Node.js v24.11.0, which is not supported by Hardhat. This can lead to unexpected behavior. See https://hardhat.org/nodejs-versions


HardhatError: HH110: Invalid JSON-RPC response received: Method Not Allowed
    at parseJsonResponse (/Users/takahiro_ogawa/dev/hardhat-multibaas-plugin/sample/node_modules/hardhat/src/internal/util/jsonrpc.ts:47:11)
    at HttpProvider._fetchJsonRpcResponse (/Users/takahiro_ogawa/dev/hardhat-multibaas-plugin/sample/node_modules/hardhat/src/internal/core/providers/http.ts:218:31)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async HttpProvider.request (/Users/takahiro_ogawa/dev/hardhat-multibaas-plugin/sample/node_modules/hardhat/src/internal/core/providers/http.ts:85:29)
    at async ChainIdValidatorProvider._getChainIdFromEthNetVersion (/Users/takahiro_ogawa/dev/hardhat-multibaas-plugin/sample/node_modules/hardhat/src/internal/core/providers/chainId.ts:33:17)
    at async ChainIdValidatorProvider._getChainId (/Users/takahiro_ogawa/dev/hardhat-multibaas-plugin/sample/node_modules/hardhat/src/internal/core/providers/chainId.ts:17:25)
    at async ChainIdValidatorProvider.request (/Users/takahiro_ogawa/dev/hardhat-multibaas-plugin/sample/node_modules/hardhat/src/internal/core/providers/chainId.ts:55:29)
    at async getSigners (/Users/takahiro_ogawa/dev/hardhat-multibaas-plugin/node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:46:16)
    at async main (/Users/takahiro_ogawa/dev/hardhat-multibaas-plugin/sample/scripts/deploy-greeter.ts:5:19)

```

daenamkim commented:
> Probably not specific to this PR but I keep getting below and am stuck 😅
> 
> ```shell
> $ run deploy:greeter
> 
> > sample@1.0.0 deploy:greeter
> > HARDHAT_NETWORK=development npx hardhat run scripts/deploy-greeter.ts
> 
> WARNING: You are currently using Node.js v24.11.0, which is not supported by Hardhat. This can lead to unexpected behavior. See https://hardhat.org/nodejs-versions
> 
> 
> HardhatError: HH110: Invalid JSON-RPC response received: Method Not Allowed
>     at parseJsonResponse (/Users/takahiro_ogawa/dev/hardhat-multibaas-plugin/sample/node_modules/hardhat/src/internal/util/jsonrpc.ts:47:11)
>     at HttpProvider._fetchJsonRpcResponse (/Users/takahiro_ogawa/dev/hardhat-multibaas-plugin/sample/node_modules/hardhat/src/internal/core/providers/http.ts:218:31)
>     at processTicksAndRejections (node:internal/process/task_queues:105:5)
>     at async HttpProvider.request (/Users/takahiro_ogawa/dev/hardhat-multibaas-plugin/sample/node_modules/hardhat/src/internal/core/providers/http.ts:85:29)
>     at async ChainIdValidatorProvider._getChainIdFromEthNetVersion (/Users/takahiro_ogawa/dev/hardhat-multibaas-plugin/sample/node_modules/hardhat/src/internal/core/providers/chainId.ts:33:17)
>     at async ChainIdValidatorProvider._getChainId (/Users/takahiro_ogawa/dev/hardhat-multibaas-plugin/sample/node_modules/hardhat/src/internal/core/providers/chainId.ts:17:25)
>     at async ChainIdValidatorProvider.request (/Users/takahiro_ogawa/dev/hardhat-multibaas-plugin/sample/node_modules/hardhat/src/internal/core/providers/chainId.ts:55:29)
>     at async getSigners (/Users/takahiro_ogawa/dev/hardhat-multibaas-plugin/node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:46:16)
>     at async main (/Users/takahiro_ogawa/dev/hardhat-multibaas-plugin/sample/scripts/deploy-greeter.ts:5:19)
> ```

@tawago  I think it works for me. 🤔 If this is the matter of steps which are confused to deploy then we might need to improve the steps later.

### Local MG setup and  `deployment-config.development.js`

<img width="926" height="556" alt="image" src="https://github.com/user-attachments/assets/80f2e1f1-444a-489e-ae8f-2e4b912b8cb2" />

###  Building a plugin
```sh
➜  hardhat-multibaas-plugin git:(775-node-24) ✗ node --version
v24.11.1
➜  hardhat-multibaas-plugin git:(775-node-24) ✗ npm install

> hardhat-multibaas-plugin@1.0.0 prepare
> husky


up to date, audited 582 packages in 2s

179 packages are looking for funding
  run `npm fund` for details

24 vulnerabilities (10 low, 6 moderate, 3 high, 5 critical)

To address issues that do not require attention, run:
  npm audit fix

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
```

### Deploying a contract

```sh
➜  sample git:(775-node-24) ✗ node --version
v24.11.1
➜  sample git:(775-node-24) ✗ npm install

up to date, audited 689 packages in 2s

134 packages are looking for funding
  run `npm fund` for details

33 vulnerabilities (18 low, 5 moderate, 5 high, 5 critical)

To address issues that do not require attention, run:
  npm audit fix

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
➜  sample git:(775-node-24) ✗ npr deploy:greeter

> sample@1.0.0 deploy:greeter
> HARDHAT_NETWORK=development npx hardhat run scripts/deploy-greeter.ts

WARNING: You are currently using Node.js v24.11.1, which is not supported by Hardhat. This can lead to unexpected behavior. See https://hardhat.org/nodejs-versions


MultiBaas: Creating contract "greeter 1.0"
MultiBaas: Creating address 0x13E35943f5F1de7C965D90C97a1A138fcC6d295a with alias "greeter"
MultiBaas: Linking contract "greeter 1.0" to address "greeter"
```

daenamkim committed:
add engines to package.json files

daenamkim committed:
update lock files

daenamkim committed:
update workflow file

dbxe committed:
npm install

dbxe commented:
`npm install` resulted in some lock file changes, can you please confirm?

Otherwise, looks good. Did a basic QA running through the sample app

```
npm install
npm run build
cd sample
npm install
npx hardhat clean
npx hardhat compile
```

```
~/gi/curvegrid/hardhat-multibaas-plugin/sample 775-node-24 *2 !2 ❯ npm run deploy:greeter  19:58:13

> sample@1.0.0 deploy:greeter
> HARDHAT_NETWORK=development npx hardhat run scripts/deploy-greeter.ts

MultiBaas: Creating contract "greeter 1.0"
MultiBaas: Creating address 0xBcB518B844994FFA10F6aCED3B5AdBe7cA2EFB4B with alias "greeter"
MultiBaas: Linking contract "greeter 1.0" to address "greeter"
~/gi/curvegrid/hardhat-multibaas-plugin/sample 775-node-24 *2 !2 ❯ npm run deploy:metacoin 20:03:26

> sample@1.0.0 deploy:metacoin
> HARDHAT_NETWORK=development npx hardhat run scripts/deploy-metacoin.ts

MultiBaas: Creating contract "convertlib 1.0"
MultiBaas: Creating address 0x5F2E93a9DFc6498466692B2Eb0Ed4c7c6ec02D98 with alias "convertlib"
MultiBaas: Linking contract "convertlib 1.0" to address "convertlib"
MultiBaas: Creating contract "metacoin 1.0"
MultiBaas: Creating address 0x1A282828dbB61617aA967Cabd72c3f3A5c203331 with alias "metacoin"
MultiBaas: Linking contract "metacoin 1.0" to address "metacoin"
~/gi/curvegrid/hardhat-multibaas-plugin/sample 775-node-24 *2 !2 ❯ npm run deploy:link:greeter

> sample@1.0.0 deploy:link:greeter
> HARDHAT_NETWORK=development npx hardhat run scripts/deploy-then-link-greeter.ts

MultiBaas: Creating address 0x1eCFf743fC1149a113EA3df7463a4177f5C20735 with alias "linked_greeter"
MultiBaas: Linking contract "greeter 1.0" to address "linked_greeter"
~/gi/curvegrid/hardhat-multibaas-plugin/sample 775-node-24 *2 !2 ❯ npm run deploy:proxy:greeter

> sample@1.0.0 deploy:proxy:greeter
> HARDHAT_NETWORK=development npx hardhat run scripts/deploy-proxied-greeter.ts

MultiBaas: Creating contract "proxied_greeter 1.0"
MultiBaas: Creating address 0x6e9d6F6DE62FF3b355699b34D9A9dd37b26568c5 with alias "proxied_greeter"
MultiBaas: Linking contract "proxied_greeter 1.0" to address "proxied_greeter"
```

daenamkim commented:
> `npm install` resulted in some lock file changes, can you please confirm?

@dbxe Thank you for testing! Can you let me know your `node --version` ? 🤔  If it is not `v24.x`, please run `nvm use` in `/blockchain`.

UPDATED:

I already updated lock files but I might be wrong. So, to ensure updating them, I ran the following commands again.

```sh
# node
nvm use
Found '/private/curvegrid/PUBLIC/hardhat-multibaas-plugin/.nvmrc' with version <24>
Now using node v24.11.1 (npm v11.6.2)

node --version
v24.11.1

# /
rm -rf node_modules
npm install

# sample
cd sample
rm -rf node_modules
npm install

# git status
git status
On branch 775-node-24
Your branch is up to date with 'origin/775-node-24'.

nothing to commit, working tree clean
```

Could take another look? 🙏

daenamkim committed:
update package-lock files

daenamkim committed:
Merge branch '775-node-24' of github.com:curvegrid/hardhat-multibaas-plugin into 775-node-24

dbxe reviewed (approved):
LGTM 

By the way, i am also using node `v24.11.1`
